hypershift-operator/controllers/nodepool/inplace.go

```diff
@@ -100,6 +141,338 @@ func (r *NodePoolReconciler) reconcileInPlaceUpgrade(ctx context.Context, hc *hy
 	// Drain.
 	// Create Namespace/RBAC/ConfigMap/Pod in guest cluster.
 	// Mark Node as Upgrading.
+
+	// This check comes after the completion, so if no updates are in progress, if a node is degraded for
+	// whatever reason, we will not know until the next update, at which point hopefully the MCD is able
+	// to reconcile
+	for _, node := range nodes {
+		if node.Annotations[MachineConfigDaemonStateAnnotationKey] == MachineConfigDaemonStateDegraded {
+			setStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+				Type:               hyperv1.NodePoolUpdatingVersionConditionType,
+				Status:             corev1.ConditionFalse,
+				Reason:             hyperv1.NodePoolInplaceUpgradeFailedConditionReason,
+				Message:            fmt.Sprintf("Node %s in nodepool degraded: %v", node.Name, node.Annotations[MachineConfigDaemonReasonAnnotationKey]),
+				ObservedGeneration: nodePool.Generation,
+			})
+			return fmt.Errorf("degraded node found, cannot progress in-place upgrade")
+		}
+	}
+
+	// Create necessary update manifests, if they do not exist
+	err = r.createInPlaceUpdateManifests(ctx, hostedClusterClient, targetConfigVersionHash, ignEndpoint, caCertBytes, tokenBytes, nodePool)
```

Comment:
can we use update or upgrade consistently everywhere?

Reply:
Using update :+1: